topostats/io.py

```diff
@@ -86,6 +103,32 @@ def write_yaml(
             LOGGER.error(exception)
 
 
+def write_config_with_comments(config: str, filename: str = "config.yaml"):
+    """
+    Create a config file, retaining the comments by writing it as a string
+    rather than using a yaml handling package.
+
+    Parameters
+    ----------
+    config: str
+        A string of the entire configuration file to be saved.
+    filename: str
+        A name for the configuration file. Can have a ".yaml" on the end.
+    """
+
+    if ".yaml" not in filename and ".yml" not in filename:
+        create_config_path = "./" + filename + ".yaml"
```

Comment:
Might be overkill using `Path.cwd()` here and under the `else:` but could.

```suggestion
        create_config_path = Path.cwd() / f"{filename}.yaml"
```